iam/snippets/src/main/java/QueryTestablePermissions.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START iam_query_testable_permissions]
+
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.cloud.iam.admin.v1.IAMClient;
+import com.google.cloud.iam.admin.v1.IAMClient.QueryTestablePermissionsPagedResponse;
+import com.google.iam.admin.v1.QueryTestablePermissionsRequest;
+import java.io.IOException;
+
+public class QueryTestablePermissions {
+  public static void main(String[] args) {
+    // TODO(developer): Replace the variable before running the sample.
+    String projectId = "your-project-id";
```

Comment:
Instead of assuming they want to list grantable roles for a project, can we ask them to input the full resource ID, like we do in the other samples (https://cloud.google.com/iam/docs/creating-custom-roles#viewing-resource-permissions)?